### PR TITLE
[OPIK-153] fix get experiment with deleted dataset

### PR DESCRIPTION
**Pull request by idoberko2**
## Details
Reproduce the issue with test and make it pass.

## Issues
OPIK-153

## Testing
E2E that covers the reported issue (getById) and additional E2E test to cover the find functionality.

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java [hunk 1]
```diff
@@ -176,11 +176,20 @@ public Mono<Experiment> getById(@NonNull UUID id) {
 
                     return Mono.zip(
                             promptService.geyVersionsCommitByVersionsIds(promptVersionIds),
-                            Mono.fromCallable(() -> datasetService.findById(experiment.datasetId(), workspaceId))
+                            Mono.defer(() -> {
+                                try {
+                                    return Mono.just(Optional.of(datasetService.findById(
+                                            experiment.datasetId(), workspaceId)));
+                                } catch (NotFoundException e) {
+                                    return Mono.just(Optional.<Dataset>empty());
```
Comment (thiagohora) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java]: What about creating another method returning Optional? So we don't need to have this exception

**Review by thiagohora — DISMISSED** [review]

File: apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java (hunk 1 above)
Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java]: I considered that option and couldn't decide which is best. If you prefer I can change this quite easily.

Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java]: OK done

File: apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java [hunk 2]
```diff
@@ -159,6 +162,17 @@ public UUID getOrCreate(@NonNull String workspaceId, @NonNull String name, @NonN
         return id;
     }
 
+    @Override
+    public Optional<Dataset> getById(UUID id, String workspaceId) {
```
Comment (thiagohora) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java]:
```suggestion
    public Optional<Dataset> getById(@NonNull UUID id, @NonNull String workspaceId) {
```

**Review by thiagohora — APPROVED** [review]

File: apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java (hunk 2 above)
Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java]: Done

**Review by thiagohora — APPROVED** [review]

**Review by thiagohora — APPROVED** [review]

**Pull request merged**